api/interface.go

```diff
@@ -230,6 +240,21 @@ func WithDialOpts(newOpts DialOpts) DialOption {
 // OpenFunc is the usual form of a function that opens an API connection.
 type OpenFunc func(*Info, DialOpts) (Connection, error)
 
+// LoginParams holds parameters login parameters.
+// User can now log in using a username and password,
+// macaroons, access token or client credentials
+// (client id and secret).
+type LoginParams struct {
+	Tag              names.Tag
+	Password         string
+	Nonce            string
+	Macaroons        []macaroon.Slice
+	AccessToken      string
+	ClientID         string
+	ClientSecret     string
```

Comment:
The LoginParams struct is a useful addition but contains all of these somewhat independent attributes; you are expected to set these but not those etc.
We can consider using an OAuthParams struct to hold these ones. But I think we should consider a separate set of Login methods. This is reinforced by the messy logic in the Login call below.